weedcoco/repo/deposit.py

```diff
@@ -88,15 +93,34 @@ def deposit_weedcoco(weedcoco_path, dataset_dir, image_dir, image_hash):
     return new_dataset_dir
 
 
-def deposit(weedcoco_path, image_dir, repository_dir):
+def retrieve_image_paths(images_path):
+    for root, directories, files in os.walk(images_path):
+        for filename in files:
+            yield (os.path.join(root, filename), filename)
+
+
+def compress_to_download(dataset_dir, deposit_id, download_dir):
+    if not os.path.isdir(download_dir):
+        os.mkdir(download_dir)
```

Comment:
Again, best to avoid race conditions. Please create a function from the code I suggested for repository_dir, e.g. `mkdir_safely`.